tensorflow/core/platform/cloud/retrying_file_system.cc

```diff
@@ -37,55 +39,70 @@ bool IsRetriable(Status status) {
   }
 }
 
-Status CallWithRetries(const std::function<Status()>& f) {
-  int attempts = 0;
+Status CallWithRetries(const std::function<Status()>& f,
+                       const initial_delay_seconds) {
+  int attempt = 0;
   while (true) {
-    attempts++;
     auto status = f();
-    if (!IsRetriable(status) || attempts >= kMaxAttempts) {
+    if (!IsRetriable(status) || attempt > kMaxAttempts) {
```

Comment:
Sorry, I think I gave you a wrong suggestion. Actually the previous >= condition was more correct here.
So if the attempt number is 1, this means we're making the second attempt. So if kMaxAttempts was 1 and attempt == kMaxAttempt, this means we should fail.